backend/src/routes/api/service/modelregistry/index.ts

```diff
@@ -1,32 +1,18 @@
-import httpProxy from '@fastify/http-proxy';
-import { KubeFastifyInstance } from '../../../../types';
-import { DEV_MODE, MODEL_REGISTRY_NAMESPACE } from '../../../../utils/constants';
-import { getParam, setParam } from '../../../../utils/proxy';
+import { MODEL_REGISTRY_NAMESPACE } from '../../../../utils/constants';
+import { proxyService } from '../../../../utils/proxy';
 
-export default async (fastify: KubeFastifyInstance): Promise<void> => {
-  fastify.register(httpProxy, {
-    upstream: '',
-    prefix: '/:name',
-    rewritePrefix: '',
-    replyOptions: {
-      // preHandler must set the `upstream` param
-      getUpstream: (request) => getParam(request, 'upstream'),
-    },
-    preHandler: (request, _, done) => {
-      const name = getParam(request, 'name');
-
-      const upstream = DEV_MODE
-        ? // Use port forwarding for local development:
-          // kubectl port-forward -n <namespace> svc/<service-name> <local.port>:<service.port>
-          `http://${process.env.MODEL_REGISTRY_SERVICE_HOST}:${process.env.MODEL_REGISTRY_SERVICE_PORT}`
-        : // Construct service URL
-          `http://${name}.${MODEL_REGISTRY_NAMESPACE}.svc.cluster.local:8080`;
-
-      // assign the `upstream` param so we can dynamically set the upstream URL for http-proxy
-      setParam(request, 'upstream', upstream);
-
-      fastify.log.info(`Proxy ${request.method} request ${request.url} to ${upstream}`);
-      done();
-    },
-  });
-};
+export default proxyService(
+  null,
+  {
+    port: 8080,
+    namespace: MODEL_REGISTRY_NAMESPACE,
+  },
+  {
+    // Use port forwarding for local development:
+    // kubectl port-forward -n odh-model-registries svc/<service-name> 8085:8080
+    host: process.env.MODEL_REGISTRY_SERVICE_HOST,
+    port: process.env.MODEL_REGISTRY_SERVICE_PORT,
+  },
+  null,
+  false,
```

Comment:
@mturley If we set this to true we will be covering part of https://issues.redhat.com/browse/RHOAIENG-9037, would you want to split that into other task and have it done right away or should we wait until we get to that story?

Reply:
@lucferbux setting this to true means Istio will become required to use the MR API, right? We can ask the backend devs to help us enable it on our shared cluster, but it means until they have that configuration automatic in the operator (https://issues.redhat.com/browse/RHOAIENG-9907) any fresh installations of model registry will be broken. Not sure if we should do that.

Reply:
Ok, we'll enable it later